File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;

        long pLong = Long.parseLong(p);
        int pLength = p.length();

        int startIndex = 0;
        while (startIndex + pLength <= t.length()) {

            String sub = t.substring(startIndex, startIndex + pLength);
            if (Long.parseLong(sub) <= pLong) {
                answer++;
            }

            startIndex++;
        }

        return answer;
    }
}

File: Kotlin/프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.kt
class Solution {
    fun solution(t: String, p: String): Int {
        var answer: Int = 0

        val pInt = p.toLong()

        for (i in 0..t.length - p.length) {

            val s = t.substring(i..i + p.length-1)

            println(s)
            if (s.toLong() <= pInt) {
                answer++
            }
        }

        return answer
    }
}